### PR TITLE
Add light theme experiments

### PR DESCRIPTION
**Pull request by subsymbolic**
Task/Issue URL: https://app.asana.com/0/361428290920652/816426578517931
Tech Design URL: https://app.asana.com/0/361428290920652/814618797160182

**Description**:
Adds light theme experiments this includes:
- Control, 50%, dark theme, no toggle
- Dark, 25%, dark theme default, theme toggle in settings
- Light, 25%, light theme default, theme toggle in settings

See the Asana task for more detail.

**Steps to test this PR**:

**Test one: Upgrading**
1. Install the app as an upgrade
1. Ensure the theme is dark and there **is no** theme toggle in Settings

**Test two: Control Group**
1. Comment out the mh & mj variants in `Variant Manager`
1. Run the app and ensure the theme is **dark**
1. Go to settings and ensure there **is no** theme toggle in Settings

**Test three: Light Theme with Switch**
1. Comment out the mg & mj variants in `Variant Manager`
1. Run the app and ensure the theme is **light**
1. Go to settings and ensure there **is** a theme toggle in Settings

**Test four: Dark Theme with Switch**
1. Comment out the mg & mh variants in `Variant Manager`
1. Run the app and ensure the theme is **dark**
1. Go to settings and ensure there **is** a theme toggle in Settings

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/global/Theming.kt [hunk 1]
```diff
@@ -25,13 +25,25 @@ import android.support.v4.content.LocalBroadcastManager
...
+import com.duckduckgo.app.statistics.VariantManager.VariantFeature.ThemeFeature.LightThemeAsDefault
 
 
-fun DuckDuckGoActivity.applyTheme(settingsDataStore: SettingsDataStore): BroadcastReceiver? {
+enum class DuckDuckGoTheme {
+    DARK,
+    LIGHT;
+}
+
+fun DuckDuckGoActivity.applyTheme(settingsDataStore: SettingsDataStore, variant: Variant): BroadcastReceiver? {
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/global/Theming.kt]: Format this file and optimize imports

File: app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt [hunk 2]
```diff
@@ -19,19 +19,19 @@ package com.duckduckgo.app.settings.db
...
     var autoCompleteSuggestionsEnabled: Boolean
 }
 
 class SettingsSharedPreferences @Inject constructor(private val context: Context) : SettingsDataStore {
 
-    override var lightThemeEnabled: Boolean
-        get() = preferences.getBoolean(KEY_LIGHT_THEME_ENABLED, false)
-        set(enabled) = preferences.edit { putBoolean(KEY_LIGHT_THEME_ENABLED, enabled) }
+    override var theme: DuckDuckGoTheme?
+        get() = preferences.getString(KEY_THEME, null)?.let { DuckDuckGoTheme.valueOf(it) }
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt]: This line feels like it's doing a lot, and it makes it hard to parse quickly.

File: app/src/main/java/com/duckduckgo/app/statistics/VariantManager.kt [hunk 3]
```diff
@@ -37,7 +43,12 @@ interface VariantManager {
         val ACTIVE_VARIANTS = listOf(
             // SERP variants - do not remove
             Variant(key = "sa", weight = 1.0, features = emptyList()),
-            Variant(key = "sb", weight = 1.0, features = emptyList())
+            Variant(key = "sb", weight = 1.0, features = emptyList()),
+
+            // Light theme experiment
+            Variant(key = "mg", weight = 2.0, features = emptyList()),
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/statistics/VariantManager.kt]: Any reason for the control group getting twice the size of the others? You might be reducing the more interesting datasets

Comment (subsymbolic) [reply app/src/main/java/com/duckduckgo/app/statistics/VariantManager.kt]: I'm performing multiple comparisons and I wanted to buckets to be the same size
* switch (mh + mj) versus no switch (mg)
* light theme as default (mh) versus dark theme as default (mj)

Comment (CDRussell) [reply app/src/main/java/com/duckduckgo/app/statistics/VariantManager.kt]: When i was analysing the results of past experiments, I found trying to weight things in multiples of others pointless; it didn't make any future calculations easier as everything was always normalised anyway.

So no need to make any changes if you're happy with the setup; but it didn't seem necessary for me.

File: app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt (hunk 2 above)
Comment (CDRussell) [reply app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt]: That's slightly better, though I'd still ultimately prefer a more verbose `if` I think. (teeny, tiny preference though! 😄)

File: app/src/main/java/com/duckduckgo/app/statistics/VariantManager.kt (hunk 3 above)
Comment (subsymbolic) [reply app/src/main/java/com/duckduckgo/app/statistics/VariantManager.kt]: My motivation is one of experimental design. It doesn't matter _much_ but I want an even spilt in comparison groups as it's the smallest number that determines how fast I can arrive at a statistically significant result. Happy to zoom to talk more about this.
